proxy_urls.md

```diff
@@ -4,57 +4,51 @@ order: 1000
-When a vendor website or URL pattern is configured in EZproxy, rules are
-applied that enable EZproxy to determine whether the user requires proxying or
-not. If the user’s browser is reporting an IP address that falls within the MIT
-range, the user does not require proxy services and EZproxy redirects the
-user’s browser directly to the target site.
+Requests made via our proxy server are seen by service providers as coming from
+within the MIT network.
+
+The proxy server has the ability to determine if a connection should be proxied.
+Typically, if the user’s browser is reporting an IP address that falls within
+the MIT range, the user does not require proxy services and EZproxy redirects
+the user’s browser directly to the service provider.
 
 If the user’s browser reports an IP address that is not within the MIT range,
-EZproxy will authenticate the user and then set up a proxied session on behalf
-of the user. Whenever a URL is constructed for content that needs to be
-accessible both on-campus and off-campus, best practice is always to route the
-URL through the proxy server and let EZproxy decide whether proxying is
-required or not.
-
-#### EZproxy configuration
-While it is good practice to let EZproxy decide when its services are required,
-it can only do this if it knows about the site in question. Asking EZproxy to
-handle a site that it does not know about will result in an error. **If URLs for
-a new site are to be constructed, it is imperative that the proxy servers be
-configured accordingly before the URLs can be handled by EZproxy.** Contact TS3
-if you need EZproxy to handle URLs for a new domain.
-
-#### How to construct a proxy URL
-The only supported way to construct a URL that requests service from
-EZproxy is this: `http://libproxy.mit.edu/login?url=<target url>`. For example:
-[http://libproxy.mit.edu/login?url=https://dx.doi.org/10.1002/2017JE005302](http://libproxy.mit.edu/login?url=https://dx.doi.org/10.1002/2017JE005302)
-
-There is an unfortunate idea floating around that the same URL could be
-constructed like this: [https://dx.doi.org.libproxy.mit.edu/10.1002/2017JE005302](https://dx.doi.org.libproxy.mit.edu/10.1002/2017JE005302)
-**Do not do this!** Because some URLs constructed like this will work, it is often
-assumed that this is acceptable practice. It is not. Such URLs will sometimes
-work and sometimes fail, and this construction is not supported by EZproxy.
+EZproxy will direct the user to authenticate via touchstone and then set up a
+proxied session on behalf of the user. 
+
+When constructing a url for IP-restricted content that needs to be accessible
+both on-campus and off-campus best practice is to use a proxy URL.
+
+#### Configuration Reqirements
+The proxy server will only proxy connections to service providers listed in its
+configuration file. A proxy URL with a target domain or hostname that is not
+known to the proxy server will result in an error. Create a fix-lib ticket if
```

Comment:
Is it possible we can query EZproxy to see if it will proxy a URL before blinding sending users there? I sometimes am writing apps in which we'd want to proxy proxy-able stuff and just send direct for not proxy-able stuff without sending the user into an error state. This is mostly curiosity.

Reply:
I thought you'd never ask.

https://wikis.mit.edu/confluence/display/LIBPROXSERV/Proxy+lookup+scripts

Reply:
Would you be opposed to adding something like:
```
If your application needs to check if a URL is proxy able before providing
links to users, there are some possible ways to do that documented: <LINK>
```